### ISSUE TITLE
select-string outputs invisibly in jobs (or invoke-command)

### ISSUE THREAD

**Issue by jszabo98** [PowerShell/PowerShell#11092]
Yes, the object is output, but the result custom format is three blank lines.

# Steps to reproduce

```powershell
start-job { echo hi | select-string hi } | receive-job -wait -auto
```

# Expected behavior

```none

hi

```

# Actual behavior

```none



```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.2.3
PSEdition                      Core
GitCommitId                    6.2.3
OS                             Darwin 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 21 20:07:39 PDT 2018; root:xnu-3789.73.14~1/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mklement0** [comment]
The problem is that the formatting data associated with instance of type `Microsoft.PowerShell.Commands.MatchInfo`  is apparently incorrectly applied to the _deserialized_ instances (`Deserialized.Microsoft.PowerShell.Commands.MatchInfo`) that are returned from the background job.

The bug still exists in PowerShell Core 7.0.0-preview.5 and is also in Windows PowerShell 5.1

The data is there, however, so a suboptimal workaround (that won't give you the same formatting) is to pipe to `-Format-Table`, for instance, or to extract the `.Line` property.

**Comment by jszabo98** [comment]
For contrast, start-threadjob in PS 6 works fine, since it doesn't serialize:
```powershell
start-threadjob { echo hi | select-string hi } | receive-job -wait -auto
```
Although in PS 7 preview 6 I get this error:
```
Select-String: Object reference not set to an instance of an object.
```

**Comment by mklement0** [comment]
Thanks, @jszabo98 - that's a separate problem, though, seemingly unrelated to output formatting; I've created #11242.

**Comment by mklement0** [comment]
It looks like this is fixed as of at least v7.3.1;  possibly, the fix for #11242 also fixed this problem; if it did, the problem probably went away in v7.1
